### ISSUE TITLE
NSTI: 3.0.2: The loading of trups is slow

### ISSUE THREAD

**Issue by varnar** [NagiosEnterprises/nsti#17]
Hello,

It is any way before show traps having possibility to select the date/time, for example:
"Last 15min", "Last 60min", "Last 4h", "Last 24h", "Today", "Last 7days" and then giving some possibility to enter date/time for loading data.

Because, if DB is already of-loaded on separate server and you have 22249 Traps Total it is going to be slow.

Thanks

**Comment by lgroschen** [comment]
Varnar, thanks for your interest!

I'm going to use something like the following: https://datatables.net/examples/plug-ins/api.html

Not sure if that's the exact way to go, but something similar where the jquery doesn't load more than what needs to be viewed by the page.  Every time you use the pagination it reloads and draws the tds, etc.  Also will inlcude the number to display at a time, sorting of all columns and those will be saved through navigation until set again and lastly it includes a search bar which I am planning on adding a regex to.  If I can get a regex in there I might be able to just allow any type of query on any column against any row and include wild cards, something like this might satisfy your request.

In the mean time have you tried using the filter feature shown here?  Just use the dropdown on the page to apply the filter to the table view:

![nsti_020215_filter](https://cloud.githubusercontent.com/assets/6422564/6003299/2ce2c746-aac0-11e4-822e-9ed674b92d40.png)

**Comment by varnar** [comment]
Hello @lgroschen 
Yes, that filter is working and it really works when all traps is already loaded :)

**Comment by lgroschen** [comment]
I'm going to leave this open as a general bug.

Hope to make NSTI only load the traps that are going to display since it should know by ID.  Then you would never have to query the whole database every time you load a page or apply a filter.  Using the table in my previous comment will help to this while also adjusting the backend that loads the trap table.

**Comment by hedenface** [comment]
Added to 3.0.3

**Comment by bjornfro** [comment]
When will 3.0.3 be available and was/is this fixed there? I have this problem where NSTI is extremely slow. I have many many thousands of SNMP traps in DB.  Guess I need to revert to old Nagtrap if this is not planned to be implemented.

**Comment by jomann09** [comment]
Are you running this in conjunction with Naigos XI or Nagios Core?

The issue is that the entire DB set is being pulled into the datatables library, from what I can understand, and is being changed in the new version.

**Comment by bjornfro** [comment]
No, not running either. Not sure what that matters. I'm talking about the NSTI web interface. 

I know the issue is the entire DB being pulled. Question is if this is planned to be adjusted anytime soon?

**Comment by jvandermeulen** [comment]
@jomann09 Also curious for the answer to the question(s) of @bjornfro

**Comment by jomann09** [comment]
I was only asking because we are planning on a new version of NSTI, which will not be pulling the whole database, but will be integrated with our products. We likely won't be updating this version of NSTI any time soon.
